### ISSUE TITLE
ELF .bss SHT_NOBITS size used in last_offset calculation

### ISSUE THREAD

**Issue by urif-c2a** [lief-project/LIEF#336]
**Describe the bug**
In LIEF::ELF::Binary::add_segment<E_TYPE::ET_EXEC>() function, Binary.tcc line 292:
When calculating last_offset for a new segment, the offset(.bss) + size(.bss) is calculated for comparison. The .bss section is NOBITS type, that is, it's size in file is 0. The sum calculated may be huge, and may point to an offset outside the file end.

**To Reproduce**
Set breakpoint in the line 292.

**Expected behavior**
I guess the NOBITS should mean zero file size for this accumulation.

**Environment (please complete the following information):**
master

**Comment by romainthomas** [comment]
I agree. I will do some tests to check if it doesn't break corner cases

**Comment by romainthomas** [comment]
Should be fixed now
